app/controllers/carts_controller.rb

```diff
@@ -0,0 +1,25 @@
+class CartsController < ApplicationController
+  def index
+    redirect_to root_path unless logged_in?
+
+    @cart = session[:cart]
+    @price = update_price
+  end
+
+  def update_price
+    price = 0
+    return price if @cart.nil?
+
+    @cart&.each do |index, item|
+      product = Product.find(index)
+      price += product.price * item['quantity']
+    end
+    price
```

Comment:
Price will be returned anyway, as the last statement that gets executed is an addition on it.

Reply:
Was my intuition as well at first, but apparently it returns the cart object without the last statement, which is why i explicitly return price.

Reply:
Good to know!